packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { I18n } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type LanguageGroup = {
+	id: string;
+	name: ( translate: I18n[ '__' ] ) => string;
```

Comment:
It feels a bit off to pass `translate` as an argument here. Did we mean to use it directly in the code? A function that returns a string should be enough IMHO.

Reply:
The language groups are defined in a static file, not inside any specific context. I'm not sure how we could have translated names without passing translate to the static file unless we changed the way language groups are defined (as in, defining them in a specific runtime context rather than just a generic module). What are your thoughts for the alternatives to make this work with translated names still?

Reply:
I'd expect that we would just use a function that returns a string. See https://github.com/Automattic/wp-calypso/blob/try/extract-language-picker/client/lib/plans/plans-list.js#L25 for example.

By going with this approach, we:

* Ensure that localization functions will be invoked as late as possible, which is necessary to be sure we've loaded all the locale data already.
* Avoid the necessity to pass a specific localization function as an argument.
* Stay flexible and not enforce a specific i18n framework. By expecting just a function that returns a string, we let the package consumer decide what i18n tools to use, vs forcing them to use a particular one.

Let me know if that makes sense.

Reply:
Yes, thank you for the explanation and example!

Reply:
@tyxla Does this also apply to localization that happens within the component itself (as in the case of the Regions label?) Would we just expect the consumer to dependency inject an i18n provider that had the signature `string => string`?

Reply:
> is it safe to import __ directly from @wordpress/i18n as long as it is invoked as late as possible, as you said (like during rendering)?

I'd say it's safe to do that. That's why we wrap them in functions, so invoking them late is a good warranty that it'll be done after the locale data has already loaded.

Note: I'm not saying it's a perfect solution. We've been limited in Calypso by that for a while now, too, and we're likely to stay with that limitation in the future.

That being said, I think it's an acceptable trade-off.